### PR TITLE
Refactor the JS frontend code

### PR DESCRIPTION
**Pull request by berggren**
## Big Javascript refactor

In order to get more maintainable code and make it possible to write frontend tests this PR refactors the Angular part of the frontend. 

This is the first part of this refactor and is basically a port of the existing functionality moving away from a central controller and more to directive focused components.

**Note**: This is the first step in this refactor, and the following issues will be addressed in a later PR:
* Tests
* Tighter scopes in some of the directives

Ref issues: #46 #66 #67  
Reviewer: @mbushkov

### CODE REVIEW

File: timesketch/ui/static/components/explore/explore-event-directive.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+         * @param index-id - Index id (in the datastore) for the timeline.
+         */
+        return {
+            restrict: 'E',
+            scope: {
+                timelineNames: '=',
+                indexId: '='
+            },
+            link: function (scope, elem, attrs) {
+                elem.text(scope.timelineNames[scope.indexId])
```
Comment (mbushkov) [inline timesketch/ui/static/components/explore/explore-event-directive.js]: This directive won't really react on changes in neither timelineNmae nor indexId. Why not use a template here: "{{ timelineNames[indexId] }}"? If this is performance critical, you can use " {{ ::timelineNames[indexId] }}"

File: timesketch/ui/static/components/explore/explore-event-directive.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+         * @param index-id - Index name for the timeline.
+         */
+        return {
+            restrict: 'A',
+            scope: {
+                timelineColors: '=',
+                indexId: '='
+            },
+            link: function (scope, elem, attrs) {
+                elem.css("background", "#" + scope.timelineColors[scope.indexId])
```
Comment (mbushkov) [inline timesketch/ui/static/components/explore/explore-event-directive.js]: This directive doesn't watch timleineColors or indexId. Is this intentional?

File: timesketch/ui/static/components/explore/explore-event.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        <td width="35px;" >
+            <i ng-click="star = !star;toggleStar()" ng-class="{true: 'fa fa-lg fa-star icon-yellow', false: 'fa fa-lg fa-star-o icon-grey'}[star]"></i>
+        </td>
+
+        <td class="event-message" ng-click="getDetail();showDetails = !showDetails">
+            <span ng-show="comment">
+                <i class="fa fa-comments"></i>
+            </span>
+            [{{::event._source.timestamp_desc}}] {{::event._source.message|limitTo:500}}
```
Comment (mbushkov) [inline timesketch/ui/static/components/explore/explore-event.html]: Is there a general principle in current timesketch design re where to use one-time bindings and when to use normal bindings?
I.e. this binding is a one-time one, but comments are rendered via normal bindings. Is there a reason why one-time bindings are not used for comments?

**Comment by mbushkov** [comment]
LGTM. Johan, awesome work! The code looks so much better now. Please see my comments in the review. None of my comments are critical, so totally fine to merge this pull request first and discuss them in the meantime.

File: timesketch/ui/static/components/explore/explore-event-directive.js (hunk 1 above)
Comment (berggren) [inline timesketch/ui/static/components/explore/explore-event-directive.js]: True, I have moved this to the template instead and removed the whole directive.

File: timesketch/ui/static/components/explore/explore-event-directive.js (hunk 2 above)
Comment (berggren) [inline timesketch/ui/static/components/explore/explore-event-directive.js]: Moved to the template instead. Directive removed.

File: timesketch/ui/static/components/explore/explore-event.html (hunk 3 above)
Comment (berggren) [inline timesketch/ui/static/components/explore/explore-event.html]: The principle is to not use normal bindings where it is not necessary :) The comments in the event directive are now fixed. Thanks for pointing this out!

**Pull request merged**